### PR TITLE
Add readyEventTimeout functionality. readyEvent hangs sometimes.

### PR DESCRIPTION
**Pull request by klodoma**
Related to #1208

When readyEvent is used an not fired, BackStopJs Hangs. 

Added a readyTimeout for readyEvent or readySelector properties. Default 30000ms. 
If the readyEvent or readySelector is not fired within this time the promises will be resolved.

### CODE REVIEW

**Review by garris — CHANGES_REQUESTED** [review]
@klodoma would you be open to renaming this parameter to `readyTimeout` such that this value will be used for either `readyEvent` or `readySelector`?

**Comment by klodoma** [comment]
Sure, that's a good idea. I'll adjust it.

**Comment by klodoma** [comment]
Ok, I am using just readyTimeout now. 

`await page.waitForSelector(scenario.readySelector);` has a default timeout of 30000ms so I was not sure if I should put the `readyTimeout` as a parameter to it. Could be an option. 

Please review.

**Comment by garris** [comment]
Yes.  Please add `readyTimeout` as a parameter to `waitForSelector`.

Also,  can you please create a test for this?  Maybe set the timeout to 5 seconds and test that scenarios behave as expected.

**Comment by klodoma** [comment]
Please check the latest changes. I've added the tests too. 

Is it normal that the reference files are not up to date? Should I update/commit them?
I am getting quite some differences:

Screenshot_77

File: core/util/runPuppet.js [hunk 1]
```diff
@@ -72,10 +72,16 @@ async function processScenarioView (scenario, variantOrScenarioLabelSafe, scenar
...
   const readyEvent = scenario.readyEvent || config.readyEvent;
-  let readyResolve, readyPromise;
+  const readyTimeout = scenario.readyTimeout || config.readyTimeout || 30000;
+  let readyResolve, readyPromise, readyTimeoutTimer;
   if (readyEvent) {
     readyPromise = new Promise(resolve => {
       readyResolve = resolve;
+      // fire the ready event after the readyTimeout
+      readyTimeoutTimer = setTimeout(() => {
+        console.error(chalk.red('*** readyEvent Timeout! ***'), scenario.url);
```
Comment (garris) [inline core/util/runPuppet.js:82]: Could we make this log more explicit?  
_ReadyEvent not detected within readyTimeout limit. (\`${readyTimeout}\` seconds)_

**Comment by garris** [comment]
Also,  when you ran the tests -- did you use the docker version of the tests?  I think those text differences will go away if you use the docker version.

eg `npm run smoke-test-docker `

File: core/util/runPuppet.js (hunk 1 above)
Comment (klodoma) [reply core/util/runPuppet.js]: Fixed.

**Comment by klodoma** [comment]
> eg `npm run smoke-test-docker `

It hangs if I leave the new tests in, but that makes sense. If I remove the new tests, it run through, there are still 2 diffs. See screenshot.

Screenshot_79

**Comment by garris** [comment]
@klodoma Yes, if those are the only two differences it's totally fine.  Thank you!

**Comment by klodoma** [comment]
> @klodoma Yes, if those are the only two differences it's totally fine. Thank you!

@garris yes, only those 2.

**Review by garris — APPROVED** [review]
LGTM

**Comment by klodoma** [comment]
Would it be possible to release this asap?

**Comment by garris** [comment]
Done

```
npm notice === Tarball Details ===
npm notice name:          backstopjs
npm notice version:       5.3.5
npm notice package size:  22.8 MB
npm notice unpacked size: 28.2 MB
npm notice shasum:        dd9c7d29d998c7d8de2214be67121343c11156eb
npm notice integrity:     sha512-+ucYnXjl+UAb/[...]IXj1AuHYwSFVg==
npm notice total files:   527
npm notice
+ backstopjs@5.3.5
```

**Comment by klodoma** [comment]
Thank you!

**Pull request merged**